app/src/androidTest/java/testapp/NavigationViewTest.java

```diff
@@ -27,11 +34,49 @@ public void onInitialization_navigationMapboxMapIsNotNull() {
   }
 
   @Test
+  @Ignore
   public void onNavigationStart_mapboxNavigationIsNotNull() {
     validateTestSetup();
 
     MapboxNavigation mapboxNavigation = getNavigationView().retrieveMapboxNavigation();
 
     assertNotNull(mapboxNavigation);
   }
+
+  @Test
+  public void testNavigationViewPerformance_DCHQ_to_DCA() {
+    validateTestSetup();
+
+    // TODO record perf metrics
+
+    DirectionsRoute testRoute = DirectionsRoute.fromJson(loadJsonFromAsset("dchq_to_dca_route.json"));
+    NavigationViewOptions navigationViewOptions = NavigationViewOptions.builder()
+      .directionsRoute(testRoute)
+      .shouldSimulateRoute(true)
+      .build();
+
+    invoke(getNavigationView(), (uiController, navigationView) -> navigationView.startNavigation(navigationViewOptions));
+
+    // TODO wait until navigation arrival to end test
+    try {
+      Thread.sleep(300000L);
```

Comment:
The correct way to hold execution of an espresso test is by using `UiController#loopMainThreadForAtLeast(300000)`. You can get a reference to a UiController as part of a ViewAction.